backend/cfg/llvmize.ml

```diff
@@ -0,0 +1,583 @@
+        "Unimplemented data item encountered in [Llvmize.typ_of_data_item]"
+
+  let pp_const_data_item ppf (d : Cmm.data_item) =
+    match d with
+    | Cdefine_symbol _ -> assert false (* cannot happen *)
+    | Cint n -> fprintf ppf "%s" (Nativeint.to_string n)
+    | Csymbol_address { sym_name; sym_global = _ } -> pp_global ppf sym_name
+    | Cint8 _ | Cint16 _ | Cint32 _ | Csingle _ | Cdouble _ | Cvec128 _
+    | Cvec256 _ | Cvec512 _ | Csymbol_offset _ | Cstring _ | Cskip _ | Calign _
+      ->
+      Misc.fatal_error
+        "Unimplemented data item encountered in [Llvmize.pp_const_data_item]"
+
+  let pp_typ_and_const ppf (d : Cmm.data_item) =
+    fprintf ppf "%a %a" Llvm_typ.pp_t (typ_of_data_item d) pp_const_data_item d
+
+  let data_decl t sym (ds : Cmm.data_item list) =
+    line t.ppf "%a = global %a { %a }" pp_global sym Llvm_typ.pp_t
+      (Llvm_typ.Struct (List.map typ_of_data_item ds))
+      (pp_print_list ~pp_sep:pp_comma pp_typ_and_const)
+      ds
+
+  let symbol_decl t sym =
+    line t.ppf "%a = global i64 0" pp_global (Cmm_helpers.make_symbol sym)
+
+  let empty_fun_decl t sym =
+    line t.ppf "define void %a() { ret void }" pp_global
+      (Cmm_helpers.make_symbol sym)
+end
+
+let current_compilation_unit = ref None
+
+let llvmir_to_assembly t =
+  (* CR-someday gyorsh: add other optimization flags and control which passes to
+     perform. *)
+  let cmd =
+    match !Oxcaml_flags.llvm_path with Some path -> path | None -> Config.asm
+  in
+  Ccomp.command
+    (String.concat " "
+       (cmd
+       :: [ "-o";
+            Filename.quote t.asm_filename;
+            "-O3";
+            "-S";
+            "-x ir";
+            Filename.quote t.llvmir_filename ]))
+
+let close_out () =
+  match !current_compilation_unit with
+  | None -> ()
+  | Some t ->
+    (* Exception raised during llvmize, keep .ll file. *)
+    Out_channel.close t.oc;
+    current_compilation_unit := None
+
+let open_out ~asm_filename ~output_prefix =
+  (* Create LLVM IR file for the current compilation unit. *)
+  let llvmir_filename = output_prefix ^ ".ll" in
+  current_compilation_unit := Some (create ~llvmir_filename ~asm_filename)
+
+let fun_attrs _t _codegen_options : string list =
+  (* CR gyorsh: translate and communicate to llvm backend *)
+  []
+
+(* Mark regs we need to allocate *)
+let collect_regs t (cfg : Cfg.t) =
+  let body_regs =
+    Cfg.fold_blocks cfg
+      ~f:(fun _ block regs ->
+        let body_regs =
+          DLL.fold_left block.body
+            ~f:(fun regs (instr : Cfg.basic Cfg.instruction) ->
+              Reg.Set.add_seq
+                (Seq.append (Array.to_seq instr.res) (Array.to_seq instr.arg))
+                regs)
+            ~init:Reg.Set.empty
+        in
+        let terminator_regs = Array.to_seq block.terminator.res in
+        Reg.Set.add_seq terminator_regs body_regs |> Reg.Set.union regs)
+      ~init:Reg.Set.empty
+  in
+  let arg_regs = Array.to_seq cfg.fun_args |> Reg.Set.of_seq in
```

Comment:
```suggestion
  let arg_regs = Reg.set_of_array cfg.fun_args in
```
(I think other computations in this function could
be simplified by using the helpers from [`Reg`](https://github.com/oxcaml/oxcaml/blob/main/backend/reg.mli#L102).)